### PR TITLE
Simplest fix for the grab crash that occurs with animated complex models

### PR DESCRIPTION
**Pull request by howard-stearns**
(e.g., clicking on air hockey props in front of a spinning windmill.)

This solution merely does a mutex around all the activity that refers to model data that are sometimes recalculated.  It matters when there are multiple threads that do picking.

I think a better solution might be to have picking done once per render-frame, but that's more complicated.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: f2a66f31065c9ed9ae7444984eb6012feb763413 available at: https://gist.github.com/76d1e83a7512fe01f152<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4912.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4912.dmg

File: libraries/render-utils/src/Model.cpp [hunk 1]
```diff
@@ -552,12 +552,13 @@ bool Model::findRayIntersectionAgainstSubMeshes(const glm::vec3& origin, const g
...
+        _mutex.lock();
         foreach(const AABox& subMeshBox, _calculatedMeshBoxes) {
 
             if (subMeshBox.findRayIntersection(origin, direction, distanceToSubMesh, subMeshFace)) {
                 if (distanceToSubMesh < bestDistance) {
                     if (pickAgainstTriangles) {
-                        if (!_calculatedMeshTrianglesValid) {
+                         if (!_calculatedMeshTrianglesValid) {
                             recalculateMeshBoxes(pickAgainstTriangles);
                         }
```
Comment (AndrewMeadows) [inline libraries/render-utils/src/Model.cpp]: Bad indentation at line 561.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 95be6052331a35d5261c58effeefe67206feba23 available at: https://gist.github.com/76d1e83a7512fe01f152<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4912.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4912.dmg

File: libraries/render-utils/src/Model.cpp [hunk 2]
```diff
@@ -736,7 +745,9 @@ void Model::renderSetup(RenderArgs* args) {
     // against. We cache the results of these calculations so long as the model hasn't been
     // simulated and the mesh hasn't changed.
     if (args && !_calculatedMeshBoxesValid) {
+        _mutex.lock();
         recalculateMeshBoxes();
+        _mutex.unlock();
     }
```
Comment (AndrewMeadows) [inline libraries/render-utils/src/Model.cpp]: Should the lock/unlock happen _inside_  recalculateMeshBoxes()?  There are other places that call that function... but maybe the others are already inside locked contexts?

**Comment by hifi-gustavo** [comment]
Build console output for commit: bb4df26690fbd482ebfc963812ded0e98509f7cd available at: https://gist.github.com/76d1e83a7512fe01f152<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4912.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4912.dmg

**Comment by howard-stearns** [comment]
FWIW, the lock isn't inside recalculateMeshBoxes because there other places that make use of the results. The lock has be around the uses, too, and if the lock were in recalculateMeshBoxes(), it would be double-locking in some cases, but not others. (Probably ok, if the mutex is defined as a recursive one, which isn't default, but it was safer to reason about this way.)

**Pull request merged**